flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/test/program/TestStep.java

```diff
@@ -50,6 +50,7 @@ enum TestKind {
         SINK_WITHOUT_DATA,
         SINK_WITH_DATA,
         SINK_WITH_RESTORE_DATA,
+        FAILING_SQL
```

Comment:
Do we want to distinguish when sql is failing?
Like parsing/validation/something else?

Reply:
I generalized the approach. But would start with validation and runtime for now. All other exception types rather hint to a wrong exception design.